### ISSUE TITLE
Improvements to e2e tests

### ISSUE THREAD

**Issue by laurentsimon** [ossf/scorecard-action#198]
It would be useful to have the following to improve troubleshooting and maintaining our e2e tests:
1. In each test repo, name the workflows as: `trigger.branch.yml`, e.g. `schedule.main.yml`
2. Give each workflow a name to make it easier to locate in the Actions tab: `Trigger - Branch` should work?
4. Add more information to the body of the issues created when the e2e test fails:
```
Repo: <repo name@branch>
(optional) Workflow name: <name of workflow>
Workflow file: <path/to/workflow.yaml>
Trigger: <trigger-name>
Branch: <branch-name>
Private repo : <true/false>
```

@naveensrinivasan wdut?

/cc @azeemshaikh38

**Comment by azeemshaikh38** [comment]
Let's prioritize (2) and (3) since we need this for debugging any failures in the upcoming Golang-based action. Since `workflow_dispatch` and `pull_request` are supposed to be experimental and not expected to have full-support in our upcoming release, let's attempt those when we plan to add support for it.

**Comment by laurentsimon** [comment]
LGTM

**Comment by azeemshaikh38** [comment]
> So let's wait until these minor changes (2 and 3) are landed to cut the release?

Yes, if you mean the Golang release. Not needed for the minor release maybe.

**Comment by laurentsimon** [comment]
OK, then we have only one item left for the minor ,which @rohankh532 is working on https://github.com/ossf/scorecard-action/issues/196
Almost there!

**Comment by laurentsimon** [comment]
An additional feature is to crate an entry only if the issue is not already crated. If the issue already exists, we can add a comment with the information for the new run.

**Comment by naveensrinivasan** [comment]
> So let's wait until these minor changes (2 and 3) are landed to cut the release?
> 
> @naveensrinivasan I don't know if you've created PRs on the other repos for these, but I think I'd like to be a reviewer. Not that I'll block anything, but mostly to follow and learn how it's setup a little more in details :-) Thanks!

Yes, I will include you as the reviewer!

**Comment by laurentsimon** [comment]
Cool, thanks. Fyi you can see the setup for SLSA in https://github.com/slsa-framework/example-package/blob/main/.github/workflows/

**Comment by naveensrinivasan** [comment]
https://github.com/ossf-tests/scorecard-action/pull/1

**Comment by naveensrinivasan** [comment]
Non-main branch
https://github.com/ossf-tests/scorecard-action-non-main-branch/blob/80fd39d8425aaa571fc06b8b4b536c6c83777eb6/.github/workflows/scorecard-golang.yml#L64

https://github.com/ossf-tests/scorecard-action-non-main-branch/blob/80fd39d8425aaa571fc06b8b4b536c6c83777eb6/.github/workflows/scorecard-main.yml#L64

**Comment by laurentsimon** [comment]
You also need to set `THIS_FILE`. I could not find a `GITHUB_` for it. Let me know if you find one

**Comment by naveensrinivasan** [comment]
> You also need to set `THIS_FILE`. I could not find a `GITHUB_` for it. Let me know if you find one

Thanks, I wasn't able to find one either. I set them manually so that it will help to debug. Let me know. If you are good merge those PR's. Thanks

**Comment by laurentsimon** [comment]
Go ahead merge. No further comment on my end.

**Comment by naveensrinivasan** [comment]
They are merged. Can we close this ticket and open another if we want any other enhancements?

**Comment by laurentsimon** [comment]
SGTM. Feel free to close and add the newly-created issue to the release issue.
